### ISSUE TITLE
objc_setAssociatedObject(… OBJC_ASSOCIATION_RETAIN_NONATOMIC) won’t compile

### ISSUE THREAD

**Issue by alakoring** [Microsoft/WinObjC#2036]
My code

```
objc_setAssociatedObject(cell.button, kAssociatedSaveKey, save, OBJC_ASSOCIATION_RETAIN_NONATOMIC);
```

generates

Error		reference to 'OBJC_ASSOCIATION_RETAIN_NONATOMIC' is ambiguous	Six Ages	C:\Users\User\Documents\SixAges\Source\App\SavedGameDataSource.m	129	

though there’s no indication of what it is trying to pick between. (And of course, this works fine on iOS, compiled by Xcode.)

**Comment by rajsesh-msft** [comment]
This is hard to say. The bridge only has one definition of this value. so this is likely coming from elsewhere.  Can you follow the instructions [here](https://github.com/Microsoft/WinObjC/wiki/FAQ#how-do-i-get-my-project-log-file-to-help-determining-the-cause-of-an-issue) to enable verbose build output and compile only the file above (ctrl + f7)?  That should give you the include paths to begin with.

**Comment by alakoring** [comment]
I wonder if it's this?

C:\Users\User\Documents\SixAges\Source\App\SavedGameDataSource.m(129,66): error : reference to 'OBJC_ASSOCIATION_RETAIN_NONATOMIC' is ambiguous
  C:\WinObjC\include\objc/runtime.h(896,2) :  note: candidate found by name lookup is '::OBJC_ASSOCIATION_RETAIN_NONATOMIC' (TaskId:15)
  C:\WinObjC\include\objc/runtime.h(896,2) :  note: candidate found by name lookup is '::OBJC_ASSOCIATION_RETAIN_NONATOMIC' (TaskId:15)

[Six Ages.log.zip](https://github.com/Microsoft/WinObjC/files/795308/Six.Ages.log.zip)

**Comment by rajsesh-msft** [comment]
@DHowett-MSFT mentioned that he has seen this before when clang modules are enabled.  This is very likely a result of this definition coming from two different modules.  You can turn clang modules off as a workaround by opening the solution properties:

![image](https://cloud.githubusercontent.com/assets/15148713/23241301/330b3c18-f927-11e6-91b7-0b84721f879a.png)

If you are using @import, you'll have to change those to #import as needed.

**Comment by alakoring** [comment]
Yes, that works (for now, I turned it off only for this source file, since I occasionally do use **@import**. Thanks for getting me on to the next issue… (I don’t consider this issue resolved, since I assume the goal is to handle modules.)

**Comment by memontic-ms** [comment]
"reference to 'OBJC_ASSOCIATION_RETAIN_NONATOMIC' is ambiguous" was fixed in PR #1515
